docs/unpackaged.rst

```diff
@@ -1,14 +1,212 @@
+* ``%%%NAMESPACED_ORG%%%`` is replaced with the package’s namespace in a namespaced org *only*, not in a managed installation. Otherwise, it remains blank.
+    .. note:: This token supports use cases where components in one unpackaged metadata bundle refer to components in another, and the dependency bundle acquires a namespace by being deployed into a namespaced org.
+* ``%%%NAMESPACE_OR_C%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org). Otherwise, it is replaced with ``c``, the generic namespace used in Lightning components.
+* ``%%%NAMESPACED_ORG_OR_C%%%`` is replaced with the package's namespace in a namespaced org *only*, not in a managed installation. Otherwise, it is replaced with ``c``, the generic namespace used in Lightning components.
+* ``%%%NAMESPACE_DOT%%%`` is replaced with the package’s namespace in any context with a namespace (such as a namespaced org or managed org) followed by a period (``.``) rather than two underscores.
+    .. note:: This token is used to construct references to packaged Record Types.
+
+..
+
+    Example: A portion of metadata from the Nonprofit Success Pack (NPSP) is stored in a subdirectory under ``unpackaged/post``, meaning it's deployed after the application metadata. This metadata updates a Compact Layout on the ``Account`` object, and references packaged metadata from the application as well as from other managed packages. To complete these tasks, this metadata requires the use of namespace tokens to represent the ``npsp`` namespace, letting CumulusCI automatically adapt the metadata to deploy into managed and unmanaged contexts.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+        <compactLayouts>
+            <fullName>NPSP_Household_Account</fullName>
+            <fields>Name</fields>
+            <fields>npo02__TotalOppAmount__c</fields>
+            <fields>%%%NAMESPACE%%%Number_of_Household_Members__c</fields>
+            <label>NPSP Household Account</label>
+        </compactLayouts>
+    </CustomObject>
+
+..
+
+    Note that only the reference to the NPSP field ``Number_of_Household_Members__c`` is tokenized. (This field is called ``npsp__Number_of_Household_Members__c`` when installed as part of the managed package.) References to NPSP's own managed package dependency, ``npo02``, are not tokenized because this metadata is always namespaced when installed.
+
+    If this metadata isn't tokenized, it fails to deploy into an org containing NPSP as a beta or released managed package.
+
+.. note:: The resolution of component references in namespaced scratch orgs and in managed installations of the same metadata are not identical. Metadata that is tokenized and deploys cleanly in a namespaced scratch org can still fail in a managed context.
```

Comment:
This seems like real inside baseball. Is it reasonable to presume that advanced devs will understand this? (Again, it's completely fine if I don't because I presume at this point that only advanced devs would be this far into the written materials.)

Reply:
This is Deep Magic. I think it's an important note but it's okay that not everyone gets it. Those who are operating with these tools (namespaced scratch orgs + managed packages) will hopefully get it.

Reply:
I think this should be fine. There's a lot of jargon, but a release engineer or advanced dev is going to be the target role for this content.